services/frontend/adapters/inbound/controller.go

```diff
@@ -0,0 +1,136 @@
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	wbhk "sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/go-logr/zerologr"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	healthcheckPort = ":8081"
+	healthcheckPath = "healthz"
+)
+
+type manifestController struct {
+	mgr                        manager.Manager
+	ctx                        context.Context
+	healthcheckClient          *http.Client
+	validationWebhookPort      int
+	validationWebhookCertDir   string
+	validationWebhookPath      string
+	validationWebhookNamespace string
+}
+
+func NewManifestController(ctx context.Context) (*manifestController, error) {
+	// lookup environment variables
+	portString, err := lookupEnv("WEBHOOK_TLS_PORT")
+	if err != nil {
+		return nil, err
+	}
+	port, err := strconv.Atoi(portString)
+	if err != nil {
+		return nil, err
+	}
+	namespace, err := lookupEnv("NAMESPACE")
+	if err != nil {
+		return nil, err
+	}
+	certDir, err := lookupEnv("WEBHOOK_CERT_DIR")
+	if err != nil {
+		return nil, err
+	}
+	webhookPath, err := lookupEnv("WEBHOOK_PATH")
+	if err != nil {
+		return nil, err
+	}
+
+	// create http.Client for healthech requests
+	client := &http.Client{}
+
+	// create ManifestController object
+	var mc = manifestController{
+		ctx:                        ctx,
+		healthcheckClient:          client,
+		validationWebhookPort:      port,
+		validationWebhookNamespace: namespace,
+		validationWebhookPath:      webhookPath,
+		validationWebhookCertDir:   certDir,
+	}
+
+	// Setup the controler manager for input-manifest
+	mc.mgr, err = ctrl.NewManager(config.GetConfigOrDie(), ctrl.Options{
+		Namespace:              mc.validationWebhookNamespace,
+		HealthProbeBindAddress: healthcheckPort,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to set up manifest-controller: " + err.Error())
+	}
+
+	// Register a healthcheck endpoint
+	if err := mc.mgr.AddHealthzCheck(healthcheckPath, healthz.Ping); err != nil {
+		return nil, err
+	}
+
+	// Register the input-manifest validation endpoint
+	hookServer := &wbhk.Server{
+		Port:    mc.validationWebhookPort,
+		CertDir: mc.validationWebhookCertDir,
+	}
+	hookServer.Register(mc.validationWebhookPath, admission.WithCustomValidator(&corev1.Secret{}, &secretValidator{}))
+	if err := mc.mgr.Add(hookServer); err != nil {
+		return nil, err
+	}
+
+	return &mc, nil
+}
+
+func (mc *manifestController) Start() {
+	crlog.SetLogger(zerologr.New(&log.Logger))
+	logger := crlog.Log
+
+	// Start manager with webhook
+	if err := mc.mgr.Start(mc.ctx); err != nil {
+		logger.Error(err, "unable to run manifest-controller")
+	}
+}
+
+func (mc *manifestController) PerformHealthCheck() error {
+	req, err := http.NewRequestWithContext(mc.ctx, "GET", fmt.Sprintf("http://127.0.0.1%s/%s", healthcheckPort, healthcheckPath), nil)
+	if err != nil {
+		return err
+	}
+	resp, err := mc.healthcheckClient.Do(req)
+	log.Debug().Msgf("health status: %s", resp.Status)
```

Comment:
I don't think this message is necessary, as it will flood the logs in debug mode.

Reply:
I've looked up this approach in other tools. I will remov it 👍